java/client/src/main/java/glide/internal/protocol/BinaryCommandRequest.java

```diff
@@ -0,0 +1,143 @@
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+/**
+ * Binary-safe command request that supports mixed String/byte[] arguments. Preserves binary data
+ * integrity while providing the same routing capabilities as CommandRequest.
+ */
+public class BinaryCommandRequest {
+    private final String commandName;
+    private final List<Object> arguments; // Can be String or byte[]
+    private final RouteInfo routeInfo;
+
+    private BinaryCommandRequest(String commandName, List<Object> arguments, RouteInfo routeInfo) {
+        this.commandName = commandName;
+        this.arguments = arguments;
+        this.routeInfo = routeInfo;
+    }
+
+    public String getCommandName() {
+        return commandName;
+    }
+
+    public List<Object> getArguments() {
+        return arguments;
+    }
+
+    public RouteInfo getRouteInfo() {
+        return routeInfo;
+    }
+
+    /** Create binary command with automatic routing */
+    public static BinaryCommandRequest auto(BinaryCommand command) {
+        return new BinaryCommandRequest(command.getType(), command.getArguments(), RouteInfo.auto());
+    }
+
+    /** Create binary command for all nodes */
+    public static BinaryCommandRequest forAllNodes(BinaryCommand command) {
+        return new BinaryCommandRequest(
+                command.getType(), command.getArguments(), RouteInfo.allNodes());
+    }
+
+    /** Create binary command for all primaries */
+    public static BinaryCommandRequest forAllPrimaries(BinaryCommand command) {
+        return new BinaryCommandRequest(
+                command.getType(), command.getArguments(), RouteInfo.allPrimaries());
+    }
+
+    /** Create binary command for random node */
+    public static BinaryCommandRequest forRandomNode(BinaryCommand command) {
+        return new BinaryCommandRequest(command.getType(), command.getArguments(), RouteInfo.random());
+    }
+
+    /** Create binary command routed by slot key */
+    public static BinaryCommandRequest forSlotKey(
+            BinaryCommand command, String slotKey, boolean preferReplica) {
+        return new BinaryCommandRequest(
+                command.getType(), command.getArguments(), RouteInfo.bySlotKey(slotKey, preferReplica));
+    }
+
+    /** Create binary command routed by slot ID */
+    public static BinaryCommandRequest forSlotId(
+            BinaryCommand command, int slotId, boolean preferReplica) {
+        return new BinaryCommandRequest(
+                command.getType(), command.getArguments(), RouteInfo.bySlotId(slotId, preferReplica));
+    }
+
+    /** Create binary command routed by address */
+    public static BinaryCommandRequest forAddress(BinaryCommand command, String host, int port) {
+        return new BinaryCommandRequest(
+                command.getType(), command.getArguments(), RouteInfo.byAddress(host, port));
+    }
+
+    /** Serialize to binary format for JNI transmission while preserving binary data */
+    public byte[] toBytes() {
```

Comment:
There looks to be alot of duplication between this and BatchRequest#writeCommand().